### PR TITLE
Simplify portions of symbol completion

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Working on this as part of "intellisense for operators/conversions".  These methods often confuse the heck out of me.  So i'm just working on making the data flow and checks clearer and simpler.

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/Completion/CompletionProviders/SymbolCompletionProviderTests.cs [hunk 1]
```diff
@@ -9585,7 +9585,7 @@ void goo()
 }" + TestResources.NetFX.ValueTuple.tuplelib_cs;
 
             // should not crash
-            await VerifyItemExistsAsync(markup, "ToString");
+            await VerifyNoItemsExistAsync(markup);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/Completion/CompletionProviders/SymbolCompletionProviderTests.cs:9588]: this test was bogus.  we shoudl not have offered completion here.  teh cleanup simplified code and ended up fixing this.

**Review by Cosifne — APPROVED** [review]
:shipit:

File: src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs [hunk 2]
```diff
@@ -239,24 +220,20 @@ private ImmutableArray<ISymbol> GetSymbolsForExpressionOrStatementContext()
             // name).  If this is the case, we do not want to filter out inaccessible locals.
             var filterOutOfScopeLocals = _filterOutOfScopeLocals;
             if (filterOutOfScopeLocals)
-            {
-                if (_context.LeftToken.Parent.IsFoundUnder<LocalDeclarationStatementSyntax>(d => d.Declaration.Type))
-                {
-                    filterOutOfScopeLocals = false;
-                }
-            }
+                filterOutOfScopeLocals = !_context.LeftToken.GetRequiredParent().IsFoundUnder<LocalDeclarationStatementSyntax>(d => d.Declaration.Type);
```
Comment (genlu) [inline src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs:223]: The original code is easier to understand for me. Can we just merge the if statements instead?

File: src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs [hunk 3]
```diff
@@ -417,102 +397,79 @@ private ImmutableArray<ISymbol> GetSymbolsOffOfConditionalReceiver(ExpressionSyn
...
-                // we shouldn't show anything
-                if (symbol.IsThisParameter() &&
-                    expression.IsInStaticContext())
+                if (symbol is IAliasSymbol alias)
+                    symbol = alias.Target;
+
+                if (symbol.Kind is SymbolKind.NamedType or SymbolKind.Namespace)
                 {
-                    return ImmutableArray<ISymbol>.Empty;
+                    // For named typed and namespaces, we flip things around.  We only want statics and not instance members.
```
Comment (genlu) [inline src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs:443]: typo: types

**Review by genlu — APPROVED** [review]
:shipit:

**Pull request merged**